src/fruity/fruity-host-session.vala

```diff
@@ -57,19 +58,34 @@ namespace Frida {
 					remove_device (id);
 				});
 
+				initial_device_ids = yield control_client.list_devices (start_cancellable);

```

Comment:
Doesn't this still leave a tiny window where a new device could be added and we'd miss it?

Reply:
possibly. would it help to start listening first and get the list from another connection?

Reply:
Yeah I think that should eliminate it!

Reply:
Actually I just thought of another option: if the daemon is single-threaded, we can actually just do a `ListDevices` right after `Listen` has completed. Once that `ListDevices` has completed we are done, and we can ignore the list of devices that it responded with.

Reply:
i have to check, i was assuming one thread pre client, but i've got to read some code (or just place logs) in usbmuxd

Reply:
LOL ok yeah usbmuxd has a single thread!
so in theory we can use the previous implementation, just on 2 separate connections 🤦

Reply:
Ah, interesting! I wouldn't be surprised if Apple's implementation is way more advanced though. But if that's the case I'm not sure if we can fingerprint the daemon very easily – perhaps there's a command we can use?

Reply:
yeah the only way i know to tell the implementations apart is maybe to trigger this original issue in a controlled way, like:

- do what we always did (which works in Apple's implementation)
- detect the underlying connection has been closed in usbmuxd case
- redo everything using 2 different connections in case the connection has been closed

(not sure how to detect that, though, but i bet it's possible)

the disadvantage of that is in the usbmuxd case will still take twice longer, but maybe the code is simpler